### PR TITLE
improvement FEMR216: Updates to better track if user was prompted to …

### PR DESCRIPTION
**Pull request by cd2499**
…screen for diabetes and whether the screening was performed. Including a new bool column in patient_encounters table to help identify. If column in table is null, user was not prompted. Otherwise, if patient was screened, value will be true. If patient was not screened, value is false.

### CODE REVIEW

**Comment by kevinzurek** [comment]
Greg, 

Everything works. Very clean pull request by the way :+1:

File: app/femr/business/services/core/IEncounterService.java [hunk 1]
```diff
@@ -116,7 +116,7 @@
      * @param userId id of the physician that screened the patient for diabetes, not null
      * @return updated patient encounter item and/or errors if they exist, or null if errors
      */
-    ServiceResponse<PatientEncounterItem> screenPatientForDiabetes(int encounterId, int userId);
+    ServiceResponse<PatientEncounterItem> screenPatientForDiabetes(int encounterId, int userId, Boolean isScreened);
```
Comment (kevinzurek) [inline app/femr/business/services/core/IEncounterService.java]: Above this method are comments that describe each parameter - we should do the same for "isScreened" as this isn't very intuitive for a future developer. These comments show up whenever you Ctrl-q this method inside IntelliJ.

**Comment by kevinzurek** [comment]
@kdunlap @AJSclyn so you guys are aware, this is how the patient_encounters table looks for the diabetes columns after a few test runs:

image
image


Line 1: prompt turned on, prompt appeared, user said yes
Line 2: prompt turned on, prompt appeared, user said no
Line 3: prompt turned on, prompt appeared, user said yes
Line 4: prompt turned off, did not appear
Line 5: prompt turned on, did not appear

I feel like this should also be documented somewhere, but i am not sure where. Maybe in the patient_encounter model?

File: app/femr/business/services/core/IEncounterService.java (hunk 1 above)
Comment (cd2499) [inline app/femr/business/services/core/IEncounterService.java]: Nice, did not know about ctrl-q. Cool stuff. ok, updated with comments. Thanks for catching that

**Comment by kdunlap** [comment]
Greg ( @cd2499 ),

Seconded on the clean pull request. 👍

**Comment by kdunlap** [comment]
https://teamfemr.atlassian.net/browse/FEMR-216

**Pull request closed without merging**